src/Session.php

```diff
@@ -1016,6 +1024,12 @@ public static function checkValidSessionId()
             );
             if ($result->current()['count'] === 0) {
                 $valid_user = false;
+            } elseif (
+                $result->current()['last_rights_update'] !== null
+                && $result->current()['last_rights_update'] > $_SESSION['glpiactiveprofile']['last_rights_update'] ?? 0
+            ) {
+                Session::changeProfile($profile_id);
```

Comment:
I think you forgot to switch to the new function.

```suggestion
                Session::reloadCurrentProfile();
```